packages/event-bus-redis/src/loaders/index.ts

```diff
@@ -0,0 +1,41 @@
+import { ConfigModule, Logger, MedusaContainer } from "@medusajs/medusa"
+import { asValue } from "awilix"
+import RealRedis from "ioredis"
+import FakeRedis from "ioredis-mock"
+
+type LoaderOptions = {
+  container: MedusaContainer
+  configModule: ConfigModule
+  logger: Logger
+}
+
+export default async ({
+  container,
+  configModule,
+  logger,
+}: LoaderOptions): Promise<void> => {
+  if (configModule.projectConfig.redis_url) {
+    // Economical way of dealing with redis clients
+    const client = new RealRedis(configModule.projectConfig.redis_url)
+    const subscriber = new RealRedis(configModule.projectConfig.redis_url)
+
+    container.register({
+      eventBusRedisClient: asValue(client),
+      eventBusRedisSubscriber: asValue(subscriber),
+    })
+  } else {
+    logger.warn(
+      "No `redis_url` provided in project config. Redis event bus will not be available."
+    )
+
+    logger.info("Using fake Redis for event bus")
```

Comment:
Suggestion: I think we shouldn't using fakeredis if we make this abstraction. Potentially we can create another module with `fakeredis` if we want, but installing and running a redis instance in order to use the `event-bus-redis` module doesn't seem like an unreasonable requirement. I think instead of using fakeredis we should exit with an appropriate error message.

Fakeredis is generally only meant for testing and shouldn't be used to run any real environment, however we've had quite a lot of comments regarding ram consumption and crashed caused by `fakeredis` from people using it wrongly since it will sometimes eat up ram until nodejs crashes.

If we want to enable testing with fakeredis I think we should create a separate testing module for that.

Reply:
This is a great point! I've removed it 💪